corehq/messaging/scheduling/scheduling_partitioned/models.py

```diff
@@ -246,10 +249,21 @@ class CaseScheduleInstanceMixin(object):
 
     @property
     @memoized
+    def case(self):
+        try:
+            return CaseAccessors(self.domain).get_case(self.case_id)
+        except (CaseNotFound, ResourceNotFound):
+            return None
+
+    @property
+    @memoized
     def recipient(self):
         if self.recipient_type == 'Self':
-            return CaseAccessors(self.domain).get_case(self.case_id)
+            return self.case
```

Comment:
yep - it was actually a bad thing for it to raise an exception before since the framework would infinitely just keep retrying it. but yeah the framework handles a None return value from a recipient lookup appropriately where it's referenced